### PR TITLE
ES6: Convert `modules/social/share-count.js`

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

Converts `modules/social/share-count.js` to ES and Flow.

## What is the value of this and can you measure success?

ES6!

## Does this affect other platforms - Amp, Apps, etc?

No.

## Screenshots

No.

## Tested in CODE?

No.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/social/share-count.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            $shortValueEls.text(shortDisplayCount);
+        });
+    }
+};
+
+const updateTooltip = (): void => {
+    $shareCountEls.attr('title', template(tooltip, counts));
+};
+
+const addToShareCount = (val: number) => {
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/social/share-count.js]: `:void`

File: static/src/javascripts/projects/common/modules/social/share-count.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+};
+let shareCount: number = 0;
+let $fullValueEls;
+let $shortValueEls;
+
+const incrementShareCount = (amount: number): void => {
+    if (amount !== 0) {
+        shareCount += amount;
+
+        const displayCount = parseInt(shareCount.toFixed(0), 10);
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/social/share-count.js]: Do we need to use `toFixed(0)` if we are subsequently doing a `parseInt`?

**Review by SiAdcock — APPROVED** [review]
Minor comments, otherwise 👍

File: static/src/javascripts/projects/common/modules/social/share-count.js (hunk 2 above)
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/social/share-count.js]: Not sure about whether we really need it - `1.5.toFixed(0) === 2` whereas `parseInt(1.5, 10) === 1`. I didn't want to change that, so kept the existing notation.

Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/social/share-count.js]: Not sure about whether we really need it - `1.5.toFixed(0) === 2` whereas `parseInt(1.5, 10) === 1`. I didn't want to change that and so kept the existing notation.

**Pull request merged**